src/bogart/AS_BAT_SplitTig_AtReadEnds.C

```diff
@@ -0,0 +1,326 @@
+  //       to catch the reads between X and Y.  endAtAfter=X and bgnBefore=Y are set. 
+  //      -Create a segment with bgnAtAfter set to Y  (as above)
+  //    Point Z will:
+  //      -Set endBefore to Z in the current segment (as above for point W)
+  //
+  //  In both examples, a final segment needs to be added to catch the reads
+  //  at the end of the tig.
+
+  //
+  //  IMPORTANT!  breakReadEnd MUST come in read order.
+  //
+  //    A  --------------------X
+  //    B   --------------------
+  //    C    -----
+  //    D     Y1------------------
+  //    E      -----
+  //    F       --------------------
+  //    G        Y2------------------
+  //    H         ----
+  //    I          -------------------Z
+  //    J           --------------------
+  //
+  //  The desired outcome is:
+  //    reads that end before X to go into         -> 1  A
+  //    reads that begin before Y1                 -> 2  B C      (C is ambiguous, either 1 or 2)
+  //    reads that begin after Y1 (but before Y2)  -> 3  D E F    (E is ambiguous, either 1, 2 or 3)
+  //    reads that begin after Y2 and end before Z -> 4  G H I    (H is ambiguous, either 1, 2, 3 or 4)
+  //    reads that end after Z                     -> 5  J
+  //
+  //  But if we sort by position:
+  //    reads that begin before Y1                 -> 1  A B C
+  //    reads that begin after Y1 but before Y2    -> 2  D E F
+  //    reads that begin after Y2 and end before X -> 3  H
+  //    reads that end before Z                    -> 4  G I
+  //    reads that end after Z                     -> 5  J
+  //
+
+  uint32     nb = 0;
+  uint32     mb = BE.size() * 2;
+
+  breakDesc *breaks = new breakDesc [mb];
+
+  for (uint32 bp=0; bp<BE.size(); bp++) {
+    if (BE[bp].breakHigh == false) {
+      breaks[nb].bgnBefore = BE[bp].splitCoord;          //  Restrict current region to reads                   ----------  <- this read!
+      breaks[nb].bgnSet    = true;                       //  that begin before the break point.                    Y----------
+
+      if (breaks[nb].endSet == true) {                   //  If there is an end point set for the current       ----------X
+        nb++;                                            //  block, make a new block to catch all the reads        ------------- <- this read!
+        breaks[nb].endAtAfter = breaks[nb-1].endBefore;  //  that end after that block but begin before the          Y-------------
+        breaks[nb].bgnBefore  = BE[bp].splitCoord;       //  block we're adding next.
+      }
+
+      nb++;                                              //  Add a new block, requiring reads
+      breaks[nb].bgnAtAfter = BE[bp].splitCoord;         //  to begin at or after the point.
+      breaks[nb].bgnSet     = true;                      //
+
+      breaks[nb].endAtAfter = BE[bp].splitCoord;         //  Remove any end restriction set below.
+    }
+
+    else {
```

Comment:
Never seen such indentation before and IMHO it's hard to read, because at that point I'm sure that if has already ended :)